build_msvc/msvc-autogen.py

```diff
@@ -42,8 +46,28 @@ def parse_makefile(makefile):
                     lib_sources[current_lib] = []
                     break
 
+def set_custom_paths(qtDir, protocPath):
+    with open(os.path.join(SOURCE_DIR, '..\\build_msvc\\Directory.build.props.in'), 'r', encoding='utf-8') as rfile:
+        s = rfile.read()
+        if qtDir:
+            s = s.replace(DEFAULT_QT_DIR, qtDir)
+        if protocPath:
+            s = s.replace(DEFAULT_PROTOC_PATH, protocPath)
+    with open(os.path.join(SOURCE_DIR, '..\\build_msvc\\Directory.build.props'), 'w', encoding='utf-8',newline='\n') as wfile:
+        wfile.write(s)
 
 def main():
+    parser = argparse.ArgumentParser(description='Bitcoin-core msbuild configuration initialiser.')
+    parser.add_argument('-qtdir', nargs='?',help='Optionally sets the directory to use for the static '
```

Comment:
Nit: Could run `yapf` on the diff to get the proper formatting? See @MarcoFalke's https://github.com/MarcoFalke/yapf-diff